haystack/nodes/file_classifier/file_type.py

```diff
@@ -34,6 +35,16 @@ def __init__(self, supported_types: List[str] = DEFAULT_TYPES):
 
         self.supported_types = supported_types
 
+    def _estimate_extension(self, file_path: Path) -> str:
+        """
+        Return the extension found based on the contents of the given file
+
+        :param file_path: the path to extract the extension from
+        """
+        extension = magic.from_file(file_path, mime=True)
+        return '.' + extension.split('.')[-1]
```

Comment:
Thank you for your answer @ZanSara !! In line 45 above my intention was to remove the elements preceding the `'/'`. For example, return `'.pdf`' instead of` 'application/pdf'` but my syntax is wrong and it should have been like this : `return '.' + extension.split('/')[-1]`. I chose to return only the last element of the extension in order to be consistent with the pre-existing code which  returns `.pdf` as well.

Reply:
Oh I see! That makes sense. A couple of things come to mind then:
- Let's lowercase the file extensions and the extension found by magic to avoid that files called `DOCUMENT.PDF` break the similarity check
- Let's add some test for the most common/interesting document types and also some other random, but common types. I could imagine a test checking txt files, .md and .rst files, pdf files, but also Word and PowerPoint documents, Libreoffice documents & other common text processors if you have any in mind, and then some image, audio and zip files. That's to make sure they're not mistaken for documents, rather than classified correctly.

Tests for FileTypeClassifier can be added here: https://github.com/deepset-ai/haystack/blob/master/test/test_filetype_classifier.py If you need help with the tests let me know :slightly_smiling_face:

Reply:
So I have made the changes that you asked me but I haven't committed them yet because I want to finish the tests first. Can you please tell me how can I run the tests locally so I can test any new method before committing ?

Reply:
Sure! You can see the complete walkthrough here: https://github.com/deepset-ai/haystack/blob/master/CONTRIBUTING.md#local but in short, you should:

- Install the development dependencies with `pip install -e .[dev]` (from haystack's root folder)

- Run the tests with `pytest test/test_filetype_classifier.py`. No need to worry about other tests for now: it's better to let the CI run them. A full test suite run would require you to setup all document stores locally, which I think is overkill given that this node use none of them.

Hope it helps. Also, do not worry about a red CI:  If any test out of `test_filetype_classifier.py` fails, I'll look into them myself :+1: (because in that case it's probably the test that's faulty).

Reply:
@ZanSara one more question please. I am also using the `mimetypes` module so I believe that I have to add it somewhere into the `setup.cfg`. Where should it go ?

Reply:
Hello! `mimetypes` should be part of the standard library (https://docs.python.org/3.9/library/mimetypes.html), so no need to add it to the dependencies 🙂